### DISCUSSION TITLE
Release build with debug symbols (RelWithDebInfo)

### DISCUSSION THREAD

**Discussion by aristotelos** [microsoft/vcpkg#38960] [Q&A] [answered]
Is there an option to build your dependencies as a release build with debug symbols (CMake `RelWithDebInfo` build type)? At the moment, debug symbols are not present for dependencies in a Linux build when building with CMAKE_BUILD_TYPE `RelWithDebInfo` and using the triplet [x64-linux-dynamic](https://github.com/microsoft/vcpkg/blob/master/triplets/community/x64-linux-dynamic.cmake).

The current Windows toolchain seems to use the `/Z7` CXX compiler flag in both cases: https://github.com/microsoft/vcpkg/blob/01f602195983451bc83e72f4214af2cbc495aa94/scripts/toolchains/windows.cmake#L71-L74

Would the best option be to specify a custom triplet with a `VCPKG_CXX_FLAGS_RELEASE` option set to `-g`?

Note: This seems related to https://github.com/microsoft/vcpkg/discussions/17921, but I would like to know possible workarounds for this issue.

**Comment by kiwixz** [comment, marked as answer]
> Would the best option be to specify a custom triplet with a VCPKG_CXX_FLAGS_RELEASE option set to -g?

Yes.  You may also need the MSVC debug runtime flag (`MTd` or `MDd`).

**Comment by aristotelos** [comment]
OK, I tried this out and it doesn't seem to work with e.g. qt5-base, but that is probably because qt5-base uses QMake and does not map settings from CMake correctly. See #39140

**Reply by aristotelos** [reply]
Closed this issue because it was due to our custom `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` that did not copy `VCPKG_CXX_FLAGS` properly as e.g. https://github.com/microsoft/vcpkg/blob/master/scripts/toolchains/linux.cmake#L79 does. I have now tested that this approach works for Linux.
